ballerina/json_api.bal

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Convert value of type `json` to subtype of `anydata`.
+#
+# + v - Source JSON value
+# + options - Options to be used for filtering in the projection
+# + t - Target type
+# + return - On success, returns the given target type value, else returns an `jsondata:Error`
```

Comment:
Can we phrase this similar to langlib functions? 

If we are using the current form, need to change things like
- value belonging to the given target type
- returns an `jsondata:Error` value


`jsondata:Error` is an identifier, so we shouldn't say just an `jsondata:Error`